recipes/cuda-histogram/meta.yaml

```diff
@@ -0,0 +1,51 @@
+{% set name = "cuda_histogram" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: a3672c474dd094e2fefd364757d8fb6aa016af6e2ffee356e2a9a78c88f58690
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  skip: True  # [win or osx or py<38]
+
+requirements:
```

Comment:
It might be worth giving https://conda-forge.org/docs/maintainer/knowledge_base/#cuda-builds a read if you haven't already. I don't have any great advice here, but it might also be worth looking at how other tools that enable CUDA support have their conda-forge builds configured.

Reply:
I don't know how to properly build CUDA enabled packages on conda-forge yet, but the failure on

```
+ pip check
cuda-histogram 0.1.0 requires cupy-cuda12x, which is not installed.
```

is coming from the [PyPI package dependency on `cupy-cuda12x`](https://github.com/scikit-hep/cuda-histogram/blob/dee1b5b97c35cae0758abf425492ddfe4d461b4d/pyproject.toml#L37-L38)

```toml
dependencies = [
  "cupy-cuda12x>=13.1.0",
...
]
```

[Nothing provides this](https://docs.cupy.dev/en/stable/install.html#installing-cupy-from-conda-forge), but I don't know how to get around the dependency check.

Reply:
@hmaarrfk :wave: Given your PR https://github.com/conda-forge/staged-recipes/pull/26116 I'm curious if you have any advice or thoughts here.

Reply:
Also cc @jakirkham  given that you added `cupy` to conda-forge in the first place (**huge** thanks there! :bow:) https://github.com/conda-forge/staged-recipes/pull/9959

Reply:
Also forgot to mention, we have some [more up-to-date docs]( https://github.com/conda-forge/cuda-feedstock/tree/main/recipe#readme )

We have an issue open to refresh the webpage docs ( https://github.com/conda-forge/conda-forge.github.io/issues/1927 ), which we are a bit behind on. Sorry about that 😞

Reply:
Ah and one more suggestion here

Regarding the CI failure. This is coming from `pip check`. Basically `pip` is saying `cupy-cuda12x` is not installed, which is true as we just install it as `cupy`. In conda, packages depend on CUDA so have this information encoded as a dependency of the package. Currently wheels lack a way to do this; so, authors have resorted to encoding this information in the wheel name (like `cupy-cuda11x` and `cupy-cuda12x`)

To fix `pip check`, would recommend [adding a patch in the recipe]( https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#patches ) to change `pyproject.toml` to use the canonical name like so. That would fix the error

```diff
 dependencies = [
-  "cupy-cuda12x>=13.1.0",
+  "cupy>=13.1.0",
  "awkward>=2.6.3",
  "numpy>=1.22.0",
  "scipy>=1.1.0",
  "hist>=2",
  "boost-histogram",
]
```